runtime/js/30_fs.js

```diff
@@ -148,46 +154,121 @@
     path,
     options = {},
   ) {
-    ops.op_remove_sync({
-      path: pathFromURL(path),
-      recursive: !!options.recursive,
-    });
+    ops.op_remove_sync(
+      pathFromURL(path),
+      !!options.recursive,
+    );
   }
 
   async function remove(
     path,
     options = {},
   ) {
-    await core.opAsync("op_remove_async", {
-      path: pathFromURL(path),
-      recursive: !!options.recursive,
-    });
+    await core.opAsync(
+      "op_remove_async",
+      pathFromURL(path),
+      !!options.recursive,
+    );
   }
 
   function renameSync(oldpath, newpath) {
-    ops.op_rename_sync({
-      oldpath: pathFromURL(oldpath),
-      newpath: pathFromURL(newpath),
-    });
+    ops.op_rename_sync(
+      pathFromURL(oldpath),
+      pathFromURL(newpath),
+    );
   }
 
   async function rename(oldpath, newpath) {
-    await core.opAsync("op_rename_async", {
-      oldpath: pathFromURL(oldpath),
-      newpath: pathFromURL(newpath),
-    });
+    await core.opAsync(
+      "op_rename_async",
+      pathFromURL(oldpath),
+      pathFromURL(newpath),
+    );
   }
 
+  // Extract the FsStat object from the encoded buffer.
+  // See `runtime/ops/fs.rs` for the encoder.
+  //
+  // This is not a general purpose decoder. There are 4 types:
+  //
+  // 1. date
+  //  offset += 4
+  //  1/0 | extra padding | high u32 | low u32
+  //  if date[0] == 1, new Date(u64) else null
+  //
+  // 2. bool
+  //  offset += 2
+  //  1/0 | extra padding
+  //
+  // 3. u64
+  //  offset += 2
+  //  high u32 | low u32
+  //
+  // 4. ?u64 converts a zero u64 value to JS null on Windows.
+  function createByteStruct(types) {
+    // types can be "date", "bool" or "u64".
+    // `?` prefix means optional on windows.
+    let offset = 0;
+    let str =
+      'const unix = Deno.build.os === "darwin" || Deno.build.os === "linux"; return {';
```

Comment:
This accesses `Deno` as a global, at runtime.
The build won't change, right?
This should be evaluated at snapshot-time, and used to construct the different string shapes, AOT, or at least passed in at snapshot-time, for correctness.

Reply:
build.os is not there at snapshot time. It's initialized after all JS source is evaluated. This probably requires a seperate refactor.